### ISSUE TITLE
Regarding E5M2

### ISSUE THREAD

**Issue by hailo231** [NVIDIA/TransformerEngine#2172]
Hi all , If i wanted to do inference using E5M2 format instead of E4M3 what recipe do i have to use? or Can i use E5M2 for inference even, (theoretically it is not acceptable )  but for benchmarking wanted to do inference using E5M2 !
Thanks

**Comment by zhongbozhu** [comment]
Technically you can, but numerically speaking using E5M2 is not as good as E4M3. 
Do you have data points for using E5M2 to achieve better performance or accuracy? Can you explain a bit more about the motivation?

**Comment by timmoon10** [comment]
If you're using PyTorch, the FP8 dtype is configurable in the recipe. For example, here is the config for FP8 current scaling: https://github.com/NVIDIA/TransformerEngine/blob/af2a0c16ec11363c0af84690cd877a59f898820e/transformer_engine/common/recipe/__init__.py#L224

To accomplish pure E5M2 training:
```python
recipe = transformer_engine.common.recipe.Float8CurrentScaling(fp8_format=transformer_engine.common.recipe.Format.E5M2)
with transformer_engine.pytorch.fp8_autocast(fp8_recipe=recipe):
    out = mymodel(data)
loss = myloss(out)
loss.backward()
```

I don't see why this wouldn't work, other than numerical/convergence issues, but be advised that unusual configurations are not well tested and are likely to break.

**Comment by ksivaman** [comment]
Actually, pure E5M2 training is unsupported and will result in a runtime failure during recipe creation ([ref](https://github.com/NVIDIA/TransformerEngine/blob/release_v2.8/transformer_engine/common/recipe/__init__.py/#L242)). We catch this restriction in TE and the underlying issue comes from the fact that `E5M2 x E5M2` GEMMs are unsupported in cublas. The hybrid recipe only uses E5M2 quantization for the incoming gradient tensor, which results in `E4M3 x E5M2` GEMM for dgrad and wgrad in the backward pass.

**Comment by sbhavani** [comment]
Closing as answered. TE does not support pure E5M2 through the standard FP8 recipes
